### PR TITLE
Allow more variables to be defined in settings

### PR DESCRIPTION
**Pull request by YesThatAllen**
Allows more of the configuration previously done in plugin.rb to be pulled from the discourse settings.

### CODE REVIEW

**Comment by g5pw** [comment]
👍 this would be very useful for us!

**Comment by YesThatAllen** [comment]
It seems that the changes in https://github.com/discourse/discourse-saml/commit/7489adcebaed6fd5aab04a3efbd18250a96e327c duplicated some of the work in this PR.

I've rebased our branch again. If this PR isn't acceptable, some feedback on it would be appreciated.

**Comment by YesThatAllen** [comment]
ok.. this has been rebased off master, and the changes broken out in to a number of discreet PRs

File: plugin.rb [hunk 1]
```diff
@@ -213,9 +213,9 @@ def index
 title = GlobalSetting.try(:saml_title) || "SAML"
 button_title = GlobalSetting.try(:saml_button_title) || GlobalSetting.try(:saml_title) || "with SAML"
 
-auth_provider title: button_title,
+auth_provider title: GlobalSetting.try(:saml_title) || button_title,
```
Comment (eviltrout) [inline plugin.rb]: `button_title` already contains `saml_title` so I don't see the point of this.

Comment (YesThatAllen) [reply plugin.rb]: `SAML_Title` resolves to "SAML" by default
`button_title` resolves to "with SAML" by default

I used this to set different options in our `app.yml` file.

Comment (YesThatAllen) [reply plugin.rb]: Of note, this PR initially added fallbacks for a number of settings.. many have been incorporated in to master so I've been rebasing this to incorporate those changes.

Agreed that at this point, there's not a huge change/improvement here, but it's still needed in our instance, and i feel can be useful for others as well.

**Comment by eviltrout** [comment]
Thank you!

**Pull request merged**